From c3a1e2d05cc1d9ddb357bacc92531bd1bebfb977 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Mon, 1 Aug 2022 17:30:36 +0200
Subject: [PATCH 1859/1936] Make sure the writer position gets reset when
 allocating new memory buffers, which may be smaller than the previous one, to
 avoid out of bounds memory writes when using the sympathetic resonance
 effect.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Effects/CombFilterBank.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Effects/CombFilterBank.cpp b/src/Effects/CombFilterBank.cpp
index 752ed983..9d247d2f 100644
--- a/src/Effects/CombFilterBank.cpp
+++ b/src/Effects/CombFilterBank.cpp
@@ -65,6 +65,8 @@ namespace zyn {
             // update mem_size and baseFreq
             mem_size = mem_size_new;
             baseFreq = baseFreqNew;
+            // reset writer position
+            pos_writer = 0;
         }
         // update nrOfStrings
         nrOfStrings = nrOfStringsNew;
-- 
2.47.0

